_posts/2016-07-04-Test-Your-Swagger-API.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Test Your Swagger API
+excerpt: Introducing support for Swagger in Dredd and more.
+layout: post
+date: 2016-07-04 00:00:00 +0100
+author: honza
+published: false
+comments: true
+---
+
+It's quite common knowledge that in Apiary we're proponents of design-first approach when it comes to building APIs. However, it's hard to do design-first and have your "single source of truth" in the API description document, if you can't easily test your API implementation against it.
+
+## Support for Swagger in Dredd
+
+It's a [long time][original-dredd-article] since we announced [Dredd][dredd], an Open Source framework for testing HTTP-based APIs. Over the years it grown into a fully-featured, mature, and very popular tool. The main flaw remaining was the fact that Dredd was able to read only [API Blueprint][api-blueprint] format.
+
+Now that's not true anymore! With 1.1.0 release, all Dredd's features are now available also for users of the [Swagger][swagger] API description format. It's an experimental support, but that's mainly because it's an initial implementation and it isn't battle-tested yet. We would like to invite everyone interested to try it out and help us to put down the "experimental" label as soon as possible.
+
+Please follow the [Quickstart][quickstart] tutorial to see how easy is to employ Dredd in your API's life cycle. Also, there's an [example application][dredd-example] you can use as a starting point
+
+## Write Dredd Hooks in Your Favorite Language
+
+Expressive power of API description formats has its limits and testing can easily get rather complex when it meets real-world situations. One of the coolest features of Dredd is its extensivity by so-called [hooks][]. Hooks are pieces of arbitrary code you can get executed before or after every test case.
+
+Originally Dredd was able to execute only hooks written in JavaScript, but in recent months it got first-class support for many other languages, namely Perl, PHP, Python, Ruby, and Go (experimental). Since there's a [tutorial][hooks-new-language] how to implement new hook handler, every now and then contributors come with support for their favorite language.
```

Comment:
I'd suggest making `Perl, PHP, Python, Ruby, and Go` individual hyperlinks to the appropriate hook documentation. I'm sure people reading this might already be using dredd and want to hear how to use their favourite language with dredd.